src/reconstruction/community.jl

```diff
@@ -0,0 +1,388 @@
+        objective_weights = repeat([1.0], inner = length(objective_mets))
+    end
+
+    if objective_column_index == 0 # needs to be created
+        nr, _ = size(community.S)
+        objcol = spzeros(nr)
+        objcol[obj_inds] .= -objective_weights
+
+        # extend model by one reaction 
+        community.S = hcat(community.S, objcol)
+        community.xl = [community.xl; 0.0]
+        community.xu = [community.xu; 1000.0]
+        community.rxns = [community.rxns; "community_biomass"]
+        community.c = spzeros(size(community.S, 2))
+        community.c[end] = 1.0
+    else # only update
+        community.S[:, objective_column_index] .= 0.0 # reset
+        community.S[obj_inds, objective_column_index] .= -objective_weights
+        dropzeros!(community.S)
+        community.c = spzeros(size(community.S, 2))
+        community.c[objective_column_index] = 1.0
+        community.xl[objective_column_index] = 0.0
+        community.xu[objective_column_index] = 1000.0
+    end
+end
+
+add_objective!(model::CoreModel, objective_met::String, objective_weight::Float64) =
+    add_objective!(model, [objective_met], [objective_weight])
+
+"""
+add_model(
+    community::CoreModel,
+    model::M,
+    exchange_rxn_ids::Vector{String},
+    exchange_met_ids::Vector{String};
+    species_name="",
+    biomass_id=""
+    ) where {M<:MetabolicModel}
+
+Add `model` to `community`, which is a pre-existing community model with
+`exchange_rxn_ids` and `exchange_met_ids`. The `species_name` is appended to
+each reaction and metabolite, see [`join`](@ref). If `biomass_id` is specified
+then a biomass metabolite for `model` is also added to the resulting model. The
+column corresponding to the `biomass_id` reaction then produces this new biomass
+metabolite with unit coefficient. Note, `exchange_rxn_ids` and
+`exchange_met_ids` must already exist in the `community` model.
+
+# Example
+```
+community = add_model(community, model, exchange_rxn_ids, exchange_met_ids; species_name="species_2", biomass_id="BIOMASS_Ecoli_core_w_GAM")
+```
+"""
+function add_model(
+    community::CoreModel,
+    model::M,
+    exchange_rxn_ids::Vector{String},
+    exchange_met_ids::Vector{String};
+    species_name = "",
+    biomass_id = "",
+) where {M<:MetabolicModel}
+
+    exchange_met_community_inds = indexin(exchange_met_ids, metabolites(community))
+    exchange_rxn_community_inds = indexin(exchange_rxn_ids, reactions(community))
+    if any(isnothing.(exchange_met_community_inds)) ||
+       any(isnothing.(exchange_rxn_community_inds))
+        throw(
+            DomainError(
+                "exchange metabolite/reaction not found.",
+                "Exchange metabolites/reactions must already be contained in the community model.",
+            ),
+        )
+    end
+
+    n_cmodel_rows, n_cmodel_cols = size(stoichiometry(community))
+    n_model_rows, n_model_cols = size(stoichiometry(model))
+    Iadd, Jadd, Vadd = findnz(stoichiometry(model))
```

Comment:
weird variable names

Reply:
what is `I`, `J`, and `V`?

Reply:
yeah, these are the default variable names used by SparseArrays from Julia, I figured we should stick to their defaults to make it simpler for someone else to read? But basically, a sparse matrix can be represented by a set of 3 arrays. Suppose `M` is some sparse matrix, then `I` = row indices, `J` = column indices and `V` = values at the associated indices. So `I[a] = i`, `J[a]=j` and then `M[i,j] = V[a]`

Reply:
can you add this to the comments?

Reply:
done